### PR TITLE
src: testing license location

### PR DESCRIPTION
**Pull request by helio-frota**
* finds the first license when LICENSE, COPYING, README are provided.
* finds the correct license when LICENSE is OK and COPYING and/or README are not.
* finds the correct license when README is OK and LICENSE and/or COPYING are not.
* finds the correct license when COPYING is OK and LICENSE and/or README are not.

### CODE REVIEW

**Comment by helio-frota** [comment]
test dir structure:
```
├── all
│   └── node_modules
│       └── sample_dependency
│           ├── COPYING
│           ├── LICENSE
│           ├── package.json
│           └── README
├── copying
│   └── node_modules
│       └── sample_dependency_copying
│           ├── COPYING
│           ├── fooLICENSE
│           ├── fooREADME
│           └── package.json
├── license
│   └── node_modules
│       └── sample_dependency_license
│           ├── fooCOPYING
│           ├── fooREADME
│           ├── LICENSE
│           └── package.json
└── readme
    └── node_modules
        └── sample_dependency_readme
            ├── fooCOPYING
            ├── fooLICENSE
            ├── package.json
            └── README
```

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
-node_modules
+/node_modules
```
Comment (danbev) [inline .gitignore]: Could you explain why this was required so I understand?

Comment (helio-frota) [inline .gitignore]: this is to ignore `licenser/node_modules` and not ignore `test/fixtures/foo/node_modules`

Comment (helio-frota) [inline .gitignore]: but, in fact I not tested without test/foo/node_modules to see if the same behavior works... I'll look on it.

Comment (helio-frota) [inline .gitignore]: seems like we need `/test/foo/node_modules`:
foo

Comment (danbev) [inline .gitignore]: Got it, thanks!

**Pull request merged**